server/server_listen.go

```diff
@@ -82,13 +85,41 @@ func (s *Server) tlsLetsEncrypt(domains []string) *tls.Config {
 	return m.TLSConfig()
 }
 
-func tlsKeyCert(key, cert string) (*tls.Config, error) {
+func (s *Server) tlsKeyCert(key, cert string, clientCaPath string) (*tls.Config, error) {
 	c, err := tls.LoadX509KeyPair(cert, key)
 	if err != nil {
 		return nil, err
 	}
-	//return file based tls config using tls defaults
-	return &tls.Config{
-		Certificates: []tls.Certificate{c},
-	}, nil
+	//mTLS requires Client CAs
+	if clientCaPath != "" {
+		files, err := ioutil.ReadDir(clientCaPath)
+		if err != nil {
+			return nil, err
+		}
+		s.Infof("Looking for client CA files from %s", clientCaPath)
+		clientCAPool := x509.NewCertPool()
+		//add all cert files from path
+		for _, file := range files {
+			f := file.Name()
+			content, err := ioutil.ReadFile(filepath.Join(clientCaPath, f))
+			if err == nil {
+				if clientCAPool.AppendCertsFromPEM(content) {
```

Comment:
Instead of reading individual files, https://golang.org/pkg/crypto/x509/#CertPool.AppendCertsFromPEM this function supports CA bundles. So you only need to call this once on a single CA file.

Reply:
please see my comment above.